### ISSUE TITLE
Use updated dperini's url regex

### ISSUE THREAD

**Issue by huan086** [ansman/validate.js#153]
dperini updated the gist

**old code**
```
          "(?!10(?:\\.\\d{1,3}){3})" +
          "(?!127(?:\\.\\d{1,3}){3})" +
          "(?!169\\.254(?:\\.\\d{1,3}){2})" +
          "(?!192\\.168(?:\\.\\d{1,3}){2})" +

...

          // path
          "(?:\\/[^\\s]*)?" +
```

**new code**
```
      "(?!(?:10|127)(?:\\.\\d{1,3}){3})" +
"(?!(?:169\\.254|192\\.168)(?:\\.\\d{1,3}){2})" +
...
    // resource path
"(?:[/?#]\\S*)?" +
```

major change is that `http://example.com?query` and `http://example.com#query` used to be detected as invalid (that's bad), while the new regex allows them

**Comment by ansman** [comment]
Can you show a case where the existing validate.js code doesn't work?

Both queries and fragments work: https://github.com/ansman/validate.js/blob/master/specs/validators/url-spec.js#L73

**Comment by huan086** [comment]
`http://example.com/path?query` works, but `http://example.com?query` doesn't. dperini's updated gist fixes that

**Comment by ansman** [comment]
Got it, thanks.

**Comment by huan086** [comment]
Oh wait, I realized that's not all.

Hostname match is now

`(?:(?:[a-z\\u00a1-\\uffff0-9]-*)*[a-z\\u00a1-\\uffff0-9]+)`

instead of

`(?:(?:[a-z\\u00a1-\\uffff0-9]+-?)*[a-z\\u00a1-\\uffff0-9]+)`

which solves the [catastrophic backtracking problem](gistcomment-1252604) by removing the `+`, and also [supports `http://g--a.com`](gistcomment-1296879)

It would be wise to update the entire regex in order to get all the fixes

**Comment by ansman** [comment]
Done

**Comment by ansman** [comment]
Released in 0.11.0
